include/gatekeeper_fib.h

```diff
@@ -195,6 +209,7 @@ struct ip_prefix {
 	const char    *str;
 	struct ipaddr addr;
 	int           len;
+	uint8_t       protocol;
```

Comment:
Adding field `protocol` to `struct ip_prefix` is the easiest solution here, but it changes the purpose of `struct ip_prefix`. Moreover, `struct ip_prefix` is already used elsewhere (e.g. requesting to flush the flow tables of GK blocks), so we should not change the meaning of `struct ip_prefix`.

It seems to me that the correct approach here is to add the parameter `protocol` to `add_fib_entry_numerical()` and pass it down to `add_fib_entry_locked()` and `init_gateway_fib_locked()`.